packages/@aws-cdk/aws-sns-subscriptions/lib/url.ts

```diff
@@ -23,17 +23,17 @@ export interface UrlSubscriptionProps {
  * @see https://docs.aws.amazon.com/sns/latest/dg/sns-http-https-endpoint-as-subscriber.html
  */
 export class UrlSubscription implements sns.ITopicSubscription {
-  constructor(private readonly url: string, private readonly props: UrlSubscriptionProps = {}) {
-    if (!url.startsWith('http://') && !url.startsWith('https://')) {
+  constructor(private readonly url: string, private readonly protocol: sns.SubscriptionProtocol, private readonly props: UrlSubscriptionProps = {}) {
+    if (!Token.isUnresolved(url) && !url.startsWith('http://') && !url.startsWith('https://')) {
       throw new Error('URL must start with either http:// or https://');
     }
   }
 
   public bind(scope: Construct, topic: sns.ITopic): void {
-    new sns.Subscription(scope, this.url, {
+    new sns.Subscription(scope, topic.node.uniqueId + 'Url', {
```

Comment:
This avoids an error:
```
Error: Cannot use tokens in construct ID: ${Token[TOKEN.11]}
```

Reply:
Ah, so it does. But now you can only add one URL subscription for any given topic.

Can you use the following expression instead?

```ts
Token.isUnresolved(this.url) ? 'UnresolvedUrl' : this.url
```

So we use the literal one if available, and only fall back to the reuse-breaking identifier if the URL is unresolved?